### ISSUE TITLE
Revisit use of 'call_in_bluesky_event_loop' 

### ISSUE THREAD

**Issue by dperl-dls** [DiamondLightSource/dodal#252]
To instantiate `ophyd_async` devices and to do any necessary synchronous actions, we are using  'bluesky.run_engine.call_in_bluesky_event_loop' which will fail if there is no active `RunEngine`. This is an issue for testing, as we can't instantiate devices on their own, and means weird stuff like having to have test fixtures in the right order so that an `RE` exists before devices are created.

## Acceptance Criteria
* There exists some useful utility function which will use `bluesky.run_engine.call_in_bluesky_event_loop' if available, but not fail if it isn't

**Comment by DominicOram** [comment]
Related to https://github.com/DiamondLightSource/hyperion/issues/1263?

**Comment by dperl-dls** [comment]
yes, when we know what the long-term solution for this will be in ophyd-async we can resolve both of these the same way
